### PR TITLE
fix: hide popover on line charts when relatedTarget is not a DOM node

### PR DESCRIPTION
**Pull request by johannes-weber**
### Description

This PR addresses the issue where an exception was thrown when leaving the popover on a chart by either:
- Focusing another element outside the page (all browsers)
- Right-clicking on a popover (Firefox only)

The root cause was that `event.relatedTarget` pointed to the `window` object in these cases. The `useMouseHover` hook has been updated to account for this scenario.

Related links, issue #, if available: AWSUI-35614

---

Why FF acts differently?
Firefox is firing the `onmousleave` event on a right click, all other browsers don't. Try it on the [mouseleave MDN page's Examples](https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseleave_event#examples). The [onmouseleave specs](https://w3c.github.io/uievents/#event-type-mouseleave) mentions
> A [user agent](https://w3c.github.io/uievents/#user-agent) MUST also dispatch this event when the element or one of its descendants moves to be no longer underneath the primary pointing device.

Not sure how the context menu is handled in this case, it could be interpreted as "no longer underneath the primary pointing device". I'm searching for references and existing FF bug reports, in case I won't find any, I'll create one.

### How has this been tested?

- Verified the fix manually in Firefox, Chrome, and Safari.
- Added unit tests to cover the `useMouseHover` hook's edge cases.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by avinashbot — COMMENTED** [review]
Do we also need changes here:
https://github.com/cloudscape-design/components/blob/main/src/area-chart/model/use-chart-model.ts#L322

**Review by avinashbot — APPROVED** [review]
Looks good at first glance, assuming all tests pass :)

File: src/mixed-line-bar-chart/hooks/use-mouse-hover.ts [hunk 1]
```diff
@@ -134,7 +134,11 @@ export function useMouseHover<T>({
   };
 
   const onPopoverLeave = (event: React.MouseEvent) => {
-    if (!isHandlersDisabled && !plotRef.current!.svg.contains(event.relatedTarget as Node)) {
+    if (
```
Comment (just-boris) [inline src/mixed-line-bar-chart/hooks/use-mouse-hover.ts:137]: How about using `nodeContains` utility, similar to this PR: https://github.com/cloudscape-design/components/pull/1374

And avoid fixing the same issue over and over again?

Comment (avinashbot) [reply src/mixed-line-bar-chart/hooks/use-mouse-hover.ts:137]: I didn't know we still had this utility - that would be a much better fix for this, yeah

Comment (just-boris) [reply src/mixed-line-bar-chart/hooks/use-mouse-hover.ts:137]: It moved to the component-toolkit, but still available and should be used to mitigate all such issues

File: src/area-chart/model/use-chart-model.ts [hunk 2]
```diff
@@ -319,11 +319,7 @@ export default function useChartModel<T extends AreaChartProps.DataTypes>({
     };
 
     const onPopoverLeave = (event: MouseEvent) => {
-      // pointing device exited from the page or from an element within the svg
-      if (
-        (event.relatedTarget !== window && plotRef.current!.svg.contains(event.relatedTarget as Node)) ||
-        interactions.get().isPopoverPinned
-      ) {
+      if (nodeContains(plotRef.current!.svg, event.relatedTarget as Element) || interactions.get().isPopoverPinned) {
```
Comment (just-boris) [inline src/area-chart/model/use-chart-model.ts:322]: I think this type-casting is not needed. I specifically updated the utility signature to accept `relatedTarget`: https://github.com/cloudscape-design/component-toolkit/blob/fa9186592934151b720c12070239ec4ded7c34a9/src/dom/node-contains.ts#L9

But thanks for using this utility anyway, I can approve this now

**Review by just-boris — APPROVED** [review]

File: src/area-chart/__tests__/area-chart-use-chart-model.test.tsx [hunk 3]
```diff
@@ -303,6 +303,47 @@ describe('useChartModel', () => {
...
+
+        expect(wrapper.findHighlightedX()?.getElement()).toHaveTextContent('1');
+
+        const mouseLeaveEvent = {
+          relatedTarget: window, // when mouse exited the page, relatedTarget is set to window.
+          clientX: 0,
+          clientY: 0,
+        } as any;
+
+        act(() => {
```
Comment (just-boris) [inline src/area-chart/__tests__/area-chart-use-chart-model.test.tsx:340]: `fireEvent` already contains `act` wrapper inside, so this is redundant

**Review by just-boris — DISMISSED** [review]

**Review by just-boris — APPROVED** [review]
